File: lingting-spring-security-web/src/test/kotlin/live/lingting/spring/security/web/SecurityAuthorizationServiceImpl.java
package live.lingting.spring.security.web;

import live.lingting.framework.security.authorize.SecurityAuthorizationService;
import live.lingting.framework.security.domain.SecurityScope;
import live.lingting.framework.security.domain.SecurityScopeAttributes;
import live.lingting.framework.security.exception.AuthorizationException;
import live.lingting.framework.security.store.SecurityStore;
import live.lingting.framework.util.LocalDateTimeUtils;
import org.springframework.stereotype.Component;

import java.time.LocalDateTime;
import java.util.Collections;
import java.util.Objects;

/**
 * @author lingting 2024-03-21 20:23
 */
@Component
public class SecurityAuthorizationServiceImpl implements SecurityAuthorizationService {

	private final SecurityStore store;

	public SecurityAuthorizationServiceImpl(SecurityStore store) {
		this.store = store;
	}

	@Override
	public SecurityScope validAndBuildScope(String username, String password) throws AuthorizationException {
		if (Objects.equals(username, password)) {
			SecurityScope scope = new SecurityScope();
			scope.setToken(username);
			scope.setTenantId(username);
			scope.setUserId(username);
			scope.setUsername(username);
			scope.setNickname(username);
			scope.setPassword(password);
			scope.setAvatar("");
			scope.setEnabled(true);
			scope.setExpireTime(LocalDateTimeUtils.toTimestamp(LocalDateTime.now().plusMonths(3)));
			scope.setRoles(Collections.singleton(username));
			scope.setPermissions(Collections.singleton(username));
			scope.setAttributes(new SecurityScopeAttributes());
			return scope;
		}
		throw new AuthorizationException("用户名或密码错误!");
	}

	@Override
	public SecurityScope refresh(String token) {
		SecurityScope scope = store.get(token);
		scope.setExpireTime(System.currentTimeMillis() * 2);
		return scope;
	}

}

File: lingting-spring-security-web/src/test/kotlin/live/lingting/spring/security/web/SecurityAuthorizationServiceImpl.kt
package live.lingting.spring.security.web

import live.lingting.framework.security.authorize.SecurityAuthorizationService
import live.lingting.framework.security.domain.SecurityScope
import live.lingting.framework.security.domain.SecurityScopeAttributes
import live.lingting.framework.security.domain.SecurityToken
import live.lingting.framework.security.exception.AuthorizationException
import live.lingting.framework.security.store.SecurityStore
import live.lingting.framework.time.DateTime
import live.lingting.framework.util.LocalDateTimeUtils.timestamp
import org.springframework.stereotype.Component

/**
 * @author lingting 2024-03-21 20:23
 */
@Component
class SecurityAuthorizationServiceImpl(private val store: SecurityStore) : SecurityAuthorizationService {

    override fun validAndBuildScope(username: String?, password: String?): SecurityScope? {
        username!!
        if (username == password) {
            val scope = SecurityScope()
            scope.authorization = username
            scope.tenantId = username
            scope.userId = username
            scope.username = username
            scope.nickname = username
            scope.password = password
            scope.avatar = ""
            scope.enabled = true
            scope.expireTime = DateTime.current().plusMonths(3).timestamp
            scope.roles = mutableSetOf<String>(username)
            scope.permissions = mutableSetOf<String>(username)
            scope.attributes = SecurityScopeAttributes()
            return scope
        }
        throw AuthorizationException("用户名或密码错误!")
    }

    override fun refresh(token: SecurityToken): SecurityScope? {
        val scope = store.get(token)
        if (scope != null) {
            scope.expireTime = System.currentTimeMillis() * 2
        }
        return scope
    }
}
